src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
```

Comment:
same as above, we are manipulating the `women` array directly, it would be a better approach to manipulate a clone of the array and return its result, following the immutability principle.

example of a better approach:

```js
const totalWoman = women.reduce((sum, woman) => sum + (woman.died - woman.born), 0);
```